ihp-hsx/Test/IHP/HSX/CustomHsxCases.hs

```diff
@@ -0,0 +1,38 @@
+{-|
+Module: IHP.HSX.TestHsx
+Description: Test helpers for HSX tests with custom tags and attributes
+-}
+module IHP.HSX.CustomHsxCases where
```

Comment:
This should be in the Test modules:

```suggestion
module Test.HSX.CustomHsxCases where
```

Reply:
So then I also create a new directory  `projectRoot/ihp-hsx/Test/HSX/`?

Reply:
Or did you mean `module Test.IHP.HSX.CustomHsxCases where`

Reply:
The `Test` modules usually have no `IHP` prefix, e.g. check out other modules like https://github.com/digitallyinduced/ihp/blob/master/Test/ModelSupportSpec.hs#L5

Reply:
Okay, I just looked at the other modes in the same directory like for example https://github.com/digitallyinduced/ihp/blob/c39d5f0ec2af03f8a2ac69ede6c377de514ee80f/ihp-hsx/Test/IHP/HSX/ParserSpec.hs#L5

But I get this error message when trying as you say:

```sh
warning: Git tree '/home/lillo/kode/ihp-contrib' is dirty
[0/1 built, 0.0 MiB DL] querying ihp-v1.3.0 on https://cache.nixos.orgdirenv: ([/nix/store/0cc1cyiifsb6g6vr2271448va439q73b-direnv-2.35.0/bin/direnv export zsh]) is taking a while to execute. Use CTRL-C to give up.
error: builder for '/nix/store/jz293rgik3qwifg36cawfkalb015i7px-ihp-hsx-v1.3.0.drv' failed with exit code 1;
       last 25 log lines:
       >     The import of ‘Data.List’ is redundant
       >       except perhaps to import instances from ‘Data.List’
       >     To import instances alone, use: import Data.List()
       >    |
       > 26 | import Data.List (foldl')
       >    | ^^^^^^^^^^^^^^^^^^^^^^^^^
       >
       > ./IHP/HSX/QQ.hs: warning: [-Wall-missed-specialisations]
       >     Could not specialise imported function ‘GHC.Exception.throw’
       >     Probable fix: add INLINABLE pragma on ‘GHC.Exception.throw’
       >
       > ./IHP/HSX/QQ.hs: warning: [-Wall-missed-specialisations]
       >     Could not specialise imported function ‘Text.Megaparsec.Error.$werrorBundlePretty’
       >       when specialising ‘Megaparsec.errorBundlePretty’
       >     Probable fix: add INLINABLE pragma on ‘Text.Megaparsec.Error.$werrorBundlePretty’
       > Preprocessing test suite 'ihp-hsx-tests' for ihp-hsx-1.4.0..
       > Building test suite 'ihp-hsx-tests' for ihp-hsx-1.4.0..
       >
       > Test/IHP/HSX/CustomHsxCases.hs:5:8: error: [GHC-28623]
       >     File name does not match module name:
       >     Saw     : ‘Test.HSX.CustomHsxCases’
       >     Expected: ‘IHP.HSX.CustomHsxCases’
       >   |
       > 5 | module Test.HSX.CustomHsxCases where
       >   |        ^^^^^^^^^^^^^^^^^^^^^^^
       For full logs, run 'nix log /nix/store/jz293rgik3qwifg36cawfkalb015i7px-ihp-hsx-v1.3.0.drv'.
error: 1 dependencies of derivation '/nix/store/9q474mz720yw67aj13j0ns07v8grvmig-ghc-9.8.2-with-packages.drv' failed to build
error: 1 dependencies of derivation '/nix/store/rlwgdxlh31y5a20y6la8ph6caisrirc2-ihp-ide-v1.3.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/5n0qp2yhx79j0iq4zjz0b096364n0qy6-ihp-v1.3.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/7fb867hfgq7fdd3x4fzxfcrcpk6a4ag1-devenv-shell-env.drv' failed to build
direnv: nix-direnv: Evaluating current devShell failed. Falling back to previous environment!
direnv: export +NIX_DIRENV_DID_FALLBACK +SMTP_ENCRYPTION +SMTP_HOST +SMTP_PORT ~PATH ~XDG_DATA_DIRS
```

Reply:
The Haskell file is in a IHP subdirectory. You need to move it from Test/IHP/HSX/CustomHsxCases.hs to Test/HSX/CustomHsxCases.hs

Reply:
ok, then I follow  :+1:

Reply:
Actually, I'm struggling. Just moving it to that directory didn't work. The the other test modules couldn't find it.

And I tried moving all of the other test files if `ihp-hsx` to that directory and it didn't work either.

So I saw that probably the cabal file needs to be fixed also:

```
test-suite ihp-hsx-tests
    type:                exitcode-stdio-1.0
    hs-source-dirs:      Test
    main-is:             Main.hs
    other-modules:
        Test.HSX.ParserSpec
        Test.HSX.QQSpec
        Test.HSX.CustomHsxCases
```


But still getting errors:

```
error: builder for '/nix/store/nhncsrwvfnqmkwv0mvpj0bdjvc90j9br-ihp-hsx-v1.3.0.drv' failed with exit code 1;
       last 25 log lines:
       >       except perhaps to import instances from ‘Data.List’
       >     To import instances alone, use: import Data.List()
       >    |
       > 26 | import Data.List (foldl')
       >    | ^^^^^^^^^^^^^^^^^^^^^^^^^
       >
       > ./IHP/HSX/QQ.hs: warning: [-Wall-missed-specialisations]
       >     Could not specialise imported function ‘GHC.Exception.throw’
       >     Probable fix: add INLINABLE pragma on ‘GHC.Exception.throw’
       >
       > ./IHP/HSX/QQ.hs: warning: [-Wall-missed-specialisations]
       >     Could not specialise imported function ‘Text.Megaparsec.Error.$werrorBundlePretty’
       >       when specialising ‘Megaparsec.errorBundlePretty’
       >     Probable fix: add INLINABLE pragma on ‘Text.Megaparsec.Error.$werrorBundlePretty’
       > Preprocessing test suite 'ihp-hsx-tests' for ihp-hsx-1.4.0..
       > Building test suite 'ihp-hsx-tests' for ihp-hsx-1.4.0..
       >
       > <no location info>: error: [GHC-82272]
       >     module ‘Test.HSX.ParserSpec’ cannot be found locally
       >
       > <no location info>: error: [GHC-82272]
       >     module ‘Test.HSX.QQSpec’ cannot be found locally
       >
       > <no location info>: error: [GHC-82272]
       >     module ‘Test.HSX.CustomHsxCases’ cannot be found locally
       For full logs, run 'nix log /nix/store/nhncsrwvfnqmkwv0mvpj0bdjvc90j9br-ihp-hsx-v1.3.0.drv'.
```

Maybe something for a separate PR?

Reply:
Ah sorry, it looks like your initial code was correct. The hsx modules were renamed a while ago when I moved them to the ihp-hsx directory https://github.com/digitallyinduced/ihp/commit/d87ce54312e1cedae069566a8507bbb374bcf34d